config/initializers/internal_api.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+# Be sure to restart your server after modifying this file.
+
+Rails.application.config.internal_api = {
+  username_path: "/var/lib/misc/infra-secrets/velum-internal-api-username",
+  password_path: "/var/lib/misc/infra-secrets/velum-internal-api-password"
+}
+
+if File.exist?(Rails.application.config.internal_api[:username_path]) &&
+    File.exist?(Rails.application.config.internal_api[:password_path])
+  Rails.application.secrets.internal_api = {
+    username: File.read(Rails.application.config.internal_api[:username_path]).strip,
+    password: File.read(Rails.application.config.internal_api[:password_path]).strip
+  }
```

Comment:
I'm fine reading the values from a text file, however I would prefer to have also a way to override them through an environment variable. I want to minimize the dependencies on files stored on the host and injected into the containers.

I think we should do something more similar to what Portus is doing.

Reply:
Right, will fix.